crates/agent/src/directives/beta_onboard.rs

```diff
@@ -36,6 +38,11 @@ pub async fn apply(
             row.catalog_prefix
         )));
     }
+    if let Some(reason) = illegal_tenant_names::ILLEGAL_TENANT_NAMES.get(&requested_tenant.as_str().to_lowercase()) {
+        return Ok(JobStatus::invalid_directive(anyhow::anyhow!(
+            "Cannot provision a new tenant with the requested name: {}.", reason
+        )));
+    }
     if agent_sql::directives::beta_onboard::is_user_provisioned(row.user_id, &mut *txn).await? {
         return Ok(JobStatus::invalid_claims(anyhow::anyhow!(
             "Cannot provision a new tenant because the user has existing grants",
```

Comment:
Subject to the block list being contained in the DB, I'd suggest updating the semantics of `agent_sql::directives::beta_onboard::tenant_exists` to use it.

We do want to leave it ambiguous as to whether a tenant is disallowed because of the block list, or because it actually already exists.